Makefile

```diff
@@ -151,29 +151,24 @@ tidy:
 		(cd $$d && go mod tidy) || exit $$? ; \
 	done
 
+.PHONY: vendor
+vendor:
+	for d in $(GO_MOD_DIRS) ; do \
+		(cd $$d && go mod vendor && go mod tidy) || exit $$? ; \
+	done
+
 .PHONY: generate
 generate: tidy
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
 	GOFLAGS='' $(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
-TMP_DIR := $(shell mktemp -d)
 .PHONY: generate-check
 generate-check: controller-gen
-	rm -rf $(TMP_DIR)
-	mkdir -p $(TMP_DIR)
-	cp -r . $(TMP_DIR)
-	cd $(TMP_DIR) && make generate
-	diff -r . $(TMP_DIR) || exit 1
-	rm -rf $(TMP_DIR)
+	./scripts/check-gittree-for-diff.sh make generate
 
 .PHONY: vendor-check
 vendor-check:
-	rm -rf $(TMP_DIR)
-	mkdir -p $(TMP_DIR)
-	cp -r . $(TMP_DIR)
-	cd $(TMP_DIR) && go mod vendor && go mod tidy
-	diff -r . $(TMP_DIR) || exit 1
-	rm -rf $(TMP_DIR)
+	./scripts/check-gittree-for-diff.sh make vendor
```

Comment:
this doesn't check go mod tidy anymore?

Reply:
We need vendor-check to make sure that we properly vendor and run go mod tidy in the right directories.

Reply:
`make vendor` includes `go mod tidy`.

However, there is a problem, namely that calling `make vendor` generates files files `{api,dpu-api}/vendor/`.

Pushed a fix.

Reply:
This is a bit confusing, I expect we do vendor in `.` and tidy in all directories where needed.